helix-core/src/comment.rs

```diff
@@ -29,12 +29,18 @@ fn find_line_comment(
                 // considered uncommented.
                 commented = false;
             }
-        } else {
-            // blank line
-            skipped.push(line);
+
+            // determine margin of 0 or 1 for uncommenting; if any comment token is not followed by a space,
+            // a margin of 0 is used for all lines.
+            if matches!(line_slice.get_char(pos + token.len()), Some(c) if c != ' ') {
```

Comment:
`token.len()` is in bytes whereas this is indexing by char. You probably need to `let token_len = token.chars().count()` outside the loop.

Reply:
I also think you can do what codemirror does and determine the margin per line (if get_char == Some(" ") then +1) https://github.com/codemirror/comment/blob/429551c1e778f1a10da5990e57dcf69c55c38fae/src/comment.ts#L189

Reply:
> I also think you can do what codemirror does and determine the margin per line (if get_char == Some(" ") then +1) https://github.com/codemirror/comment/blob/429551c1e778f1a10da5990e57dcf69c55c38fae/src/comment.ts#L189

Would that cause the margin to expand until encountering a non-space character?